### DISCUSSION TITLE
Air gapped installations with terraform

### DISCUSSION THREAD

**Discussion by rupelu** [fluxcd/flux2#1692] [Q&A]
Would it be possible to make an air gapped installation with terraform and the datasource **flux_install**. From my understanding it is currently downloading the required manifests from the internet. In some situations we do not have internet with our terraform installations. Could the manifests be bundled for such a case into the provider? So that it would have a flag to use the bundled manifests?
Our current solution is to download and package the manifests for offline installations. However we did not figure out a good way to do this automatically. We tried with some **local_file** handling but always ended up in the **kubectl_manifest** resource not knowing the count. I guess this is always the case, when we have also resource (not just datasources) in between the **flux_install** and the **kubectl_manifest**.

**Comment by stefanprodan** [comment]
If we manage to implement this: https://github.com/fluxcd/terraform-provider-flux/issues/183  then the Flux Kubernetes objects will be embedded in the provider code.
